TLM/TMPE.API/Manager/ITrafficLightManager.cs

```diff
@@ -2,5 +2,23 @@ namespace TrafficManager.API.Manager {
     using TrafficManager.API.Traffic.Enums;
 
     public interface ITrafficLightManager {
+        /// <summary>
+        /// returns if the node as any kind of traffic light (vanilla, manual, timed).
+        /// </summary>
+        public bool HasTrafficLight(ushort nodeId);
```

Comment:
```suggestion
        bool HasTrafficLight(ushort nodeId);
```
implicitly public so redundant

Reply:
@krzychu124 no that does not work
![image](https://user-images.githubusercontent.com/26344691/170862556-0954c8fa-d37a-41e8-bc99-df1169456ac3.png)
Interface Implementation has to be public.

unless if I do like this:
![image](https://user-images.githubusercontent.com/26344691/170862612-df941e35-2c61-4b9b-9e5c-1017a8f7a00c.png)

But that is not necessary in this case.

Reply:
I might not be a bad idea to implicitly define all API implementations that we do not want to use internally. If we want to go that route we need to have a discussion about it and apply the rule consistently everywhere.

Reply:
you get an error probably because you are explicitly implementing the interface and implementation you shown is private then making it explicit you force us to cast to the interface to use that method (is there another one with the same signature?). Explicit and implicit implementation are completely different things with some implications.
Basically when you create explicit implementation you won't be able to use it like this
`TraffLightManager.Instance.CanToggleTrafficLight()` (assuming that `Instance` returns `TrafficLightManager`, not the interface) 
but forced to do this instead: 
`((ITrafficLightManager)TraffLightManager.Instance).CanToggleTrafficLight()`

Reply:
What you said first does not work:
```suggestion
        bool HasTrafficLight(ushort nodeId);
```

but this will work:
```suggestion
        bool ITrafficLightManager.HasTrafficLight(ushort nodeId);
```

I understand how implicit/explicit works.